### PR TITLE
document disk space expansion and contraction

### PR DESCRIPTION
**Pull request by laverya**
OpenEBS is easy!

Rook-Ceph is *haaaaard*.

### CODE REVIEW

**Review by ricardomaraschini — APPROVED** [review]

File: src/markdown-pages/install-with-kurl/reducing-storage.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+OpenEBS LocalPV consumes a host's storage at `/var/openebs/local`.
+If you increase the amount of storage available here, OpenEBS will use it.
+If it shrinks, OpenEBS will have less storage to use.
+
+# Rook-Ceph
+
+## Expanding Storage
+
+Rook does not support expanding the storage of existing block devices, only adding new ones.
+However, a 100GB block device added to a node that already had a 100GB block device used by rook will be treated similarly to a freshly created instance with a single 200GB disk.
```
Comment (chris-sanders) [inline src/markdown-pages/install-with-kurl/reducing-storage.md:25]: I'm not sure if we want to go into detail here, this is about changing sizes but it's worth considering.

Rook assumes disk, or at least machines, have the same amount of storage available. If you for example add a 100GB disk to 1 node and not the other 2 you won't get any extra storage. If you add 100GB to one machine and 200GB to the other two, you'll get ~100GB of usable additional storage.

Maybe here we should just state that all machines should have the same amount of 'total storage' committed to Rook or the overall available storage will be less than the physical stroage.

File: src/markdown-pages/install-with-kurl/reducing-storage.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+## Contracting Storage
+
+Rook does not support shrinking a block device once it has been allocated to Rook.
+However, entire block devices can be removed from the cluster.
+The procedure to do so is long and painful, but it does exist.
```
Comment (chris-sanders) [inline src/markdown-pages/install-with-kurl/reducing-storage.md:31]: While humorous probably not appropriate if it's public facing.

File: src/markdown-pages/install-with-kurl/reducing-storage.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+## Expanding Storage
+
+Rook does not support expanding the storage of existing block devices, only adding new ones.
+However, a 100GB block device added to a node that already had a 100GB block device used by rook will be treated similarly to a freshly created instance with a single 200GB disk.
+
+## Contracting Storage
+
+Rook does not support shrinking a block device once it has been allocated to Rook.
+However, entire block devices can be removed from the cluster.
```
Comment (chris-sanders) [inline src/markdown-pages/install-with-kurl/reducing-storage.md:30]: Should we make a note here, or maybe it's done below, that you need to keep the minimum of 3 nodes with sufficient storage for your current usage?

File: src/markdown-pages/install-with-kurl/reducing-storage.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+      ROOK_BLOCK_PATH:              /dev/sdb
+      ROOK_BLOCK_PATH:              /dev/sdb
+```
+From this we can tell that the OSD using /dev/sdb is OSD 0.
+
+### Draining OSDs
+Once this is known, there are many commands to be run within the rook-ceph-tools deployment.
+You can get a shell to this deployment with `kubectl exec -it -n rook-ceph deployment/rook-ceph-tools -- bash`.
+
+Once you have determined the OSD to be removed, you can tell ceph to relocate data off of this disk with `ceph osd reweight 1 0`.
```
Comment (chris-sanders) [inline src/markdown-pages/install-with-kurl/reducing-storage.md:77]: Could be a little jarring switching from osd.2 above to osd.1 here at the very least clarifying that you're trying to remove osd.1 in this line would be helpful if someone just did the 'easy' identification above and then jumped to here.

File: src/markdown-pages/install-with-kurl/reducing-storage.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+MIN/MAX VAR: 0.78/1.30  STDDEV: 5.34
+```
+
+And additionally run `ceph osd safe-to-destroy osd.<num>` to ensure that ceph really is done with the drive:
+```
+[rook@rook-ceph-tools-6fb84b545-js4rg /]$ ceph osd safe-to-destroy osd.1
+OSD(s) 1 are safe to destroy without reducing data durability.
+```
+
+### Disabling OSDs
```
Comment (chris-sanders) [inline src/markdown-pages/install-with-kurl/reducing-storage.md:173]: I love this test step!

**Review by chris-sanders — CHANGES_REQUESTED** [review]
Just a couple of comments inline

File: src/markdown-pages/install-with-kurl/reducing-storage.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
+---
+path: "/docs/install-with-kurl/changing-storage"
```
[the remark this answers is not in the document]
Comment (laverya) [reply src/markdown-pages/install-with-kurl/reducing-storage.md:2]: yeah can do

File: src/markdown-pages/install-with-kurl/reducing-storage.md (hunk 1 above)
Comment (laverya) [reply src/markdown-pages/install-with-kurl/reducing-storage.md:25]: I think this depends on how many machines you have

If you have three machines, it's accurate - but if you have 5 or 1 I don't think it is

File: src/markdown-pages/install-with-kurl/reducing-storage.md (hunk 4 above)
Comment (laverya) [reply src/markdown-pages/install-with-kurl/reducing-storage.md:77]: where do I mention osd.2?

I can change that too

File: src/markdown-pages/install-with-kurl/reducing-storage.md (hunk 1 above)
Comment (emosbaugh) [reply src/markdown-pages/install-with-kurl/reducing-storage.md:25]: Then should you explain that replication is set to 2 or 3 and how that affects expanding disks?

File: src/markdown-pages/install-with-kurl/reducing-storage.md (hunk 4 above)
Comment (chris-sanders) [reply src/markdown-pages/install-with-kurl/reducing-storage.md:77]: osd.2 is the example above for the "simple" way of finding the disk. I agree if it were just consistently osd.1 the whole time that would also address this.

**Review by chris-sanders — APPROVED** [review]
lgtm

**Pull request merged**